compleet-client/src/bindings/fn/fn.rs

```diff
@@ -16,7 +16,16 @@ pub fn has(lua: &Lua, feature: &str) -> LuaResult<bool> {
     Ok(bit == 1)
 }
 
-/// Binding to `vim.fn.screenrow`.
-pub fn screenrow(lua: &Lua) -> LuaResult<u16> {
-    self::r#fn(lua)?.get::<_, LuaFunction>("screenrow")?.call(())
+/// Binding to `vim.fn.screenpos
+pub fn screenpos(lua: &Lua, row: u16, col: u16) -> LuaResult<(u16, u16)> {
```

Comment:
This is used in `floater.rs` to get the column and in `menu.rs` to get the row. Wouldn't it be better to use `vim.fn.screenrow` and `vim.fn.screencol` instead since we only care about either the row *or* the column, but never both at the same time?

Reply:
Neovim docs state that both of these functions are for "testing" also for some reason I always get 1 as output when I invoke both on my neovim. I can switch to using `vim.fn.winline` and `vim.fn.wincol` as recommended by the docs though.

Reply:
`winline` and `wincol` are different from `screenrow` and `screencol`: they are relative to the current window, not the whole terminal screen.

Reply:
Right. I'll switch to `screencol` and `screenrow`. On your machine, when you try to output `screencol()` or `screenrow()`, do you get `1` as a result?

Reply:
Mh, looks like I do as well. Not sure why, I think it worked the last time I tried it. `screenpos` it is I guess.

Reply:
You get `1` if you call the function manually. Try setting `:au TextChangedI * echo screencol()`, you'll see the cursor screen column as you type.

Reply:
Hmm, the results don't seem to be accurate when you press `<cr>` for example

Reply:
Doesn't matter, changing line gets caught here https://github.com/noib3/nvim-compleet/blob/191ea9dcbb728dff8c7e42a4554633b73730b4da/compleet-client/src/channel/on_bytes.rs#L37

Reply:
This seems kinda janky to me. We would have to create `state.row`, `state.col` for example, create an autocommand in `augroup.set` and update the state values in the autocommand right? Or do you have another method?

Reply:
I'm not sure what you mean? If the user presses `CR` the following call of `on_bytes` will return early (and the autocmd in `CursorMovedI` will clear the UI). The current row and column are already tracked in `State.cursor.{row, bytes}`, and their values are updated in `on_bytes`.

Reply:
I'm a bit confused. Would you prefer to use `screenrow` and `screencol` for the calculation of `colbefore`, `colafter`, `rowabove`, `rowbelow` or should I keep the current implementation with `screenpos`?

Reply:
I think using `screenrow` and `screencol` makes more sense.

Reply:
But the only way to get the expected result with `screenrow` and `screencol` would be through a `TextChangedI` autocommand. I'm a bit unsure how to implement that for the calculation of available space.

Reply:
> But the only way to get the expected result with `screenrow` and `screencol` would be through a `TextChangedI` autocommand

No? `screenrow` and `screencol` work just fine even outside of autocommands. I posted the example w/ `TextChangedI` just to show that you were always getting `1` because you were calling them manually.

Reply:
oh, got it.